src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,436 @@
+  read_value(read(io, Float64))  # x
+  read_value(read(io, Float64))  # y
+  
+  # Skip Z if present in WKB
+  if (wkb_type & WKB_Z) != 0
+    read_value(read(io, Float64))
+  end
+  
+  # Skip M if present in WKB (we never use it)
+  if (wkb_type & WKB_M) != 0
+    read_value(read(io, Float64))
+  end
+end
+
+# Convert coordinate buffer to CRS coordinates
+function create_crs_coord(buff::Tuple, crs_type::Type{<:CRS})
+  if crs_type <: LatLon
+    crs_type(buff[2], buff[1])  # LatLon(lat, lon) from WKB(x=lon, y=lat)
+  elseif crs_type <: LatLonAlt
+    crs_type(buff[2], buff[1], buff[3])  # LatLonAlt(lat, lon, alt)
+  else
+    crs_type(buff...)  # fallback for other coordinate systems
+  end
+end
+
+function parse_wkb_point(io::IOBuffer, parse_coords::Function, crs_type)
+  buff = parse_coords(io)
+  # Validate coordinates
+  if isnan(buff[1]) || isnan(buff[2])
+    error("Invalid coordinates: NaN values found")
+  end
+  coord = create_crs_coord(buff, crs_type)
+  return Point(coord)
+end
+
+# Batch parse multiple points efficiently
+function parse_wkb_points(io::IOBuffer, n::Int, parse_coords::Function, crs_type::Type{<:CRS})
+  # Read n coordinate buffers and convert to points
+  map(1:n) do _
+    buff = parse_coords(io)
+    coord = create_crs_coord(buff, crs_type)
+    Point(coord)
+  end
+end
+
+function parse_wkb_linestring(io::IOBuffer, parse_coords::Function, crs_type; is_ring::Bool=false)
+  num_points = Int(ltoh(read(io, UInt32)))
+  
+  # Use batch processing
+  points = parse_wkb_points(io, num_points, parse_coords, crs_type)
+  
+  # Check if WKB writer included duplicate closing point (varies by implementation)
+  has_equal_ends = length(points) >= 2 && points[1] == points[end]
+  
+  # Two-flag logic explained:
+  # - is_ring: Semantic flag indicating this linestring is part of a polygon structure
+  # - has_equal_ends: WKB format flag indicating if writer included duplicate closing point
+  # We need both because polygon rings should always return Ring regardless of WKB format variation
+  if is_ring || has_equal_ends
+    # Return a Ring, exclude duplicate last point if present
+    if has_equal_ends
+      return Ring(points[1:end-1]...)
+    else
+      return Ring(points...)
+    end
+  else
+    # It's an open linestring, return a Rope
+    return Rope(points...)
+  end
+end
+
+function parse_wkb_polygon(io::IOBuffer, parse_coords::Function, crs_type)
+  num_rings = Int(ltoh(read(io, UInt32)))
+  
+  # Parse exterior ring using ring parsing function
+  exterior_ring = parse_wkb_linestring(io, parse_coords, crs_type; is_ring=true)
+  
+  # Parse interior rings (holes) using map-do syntax
+  hole_rings = map(2:num_rings) do ring_idx
+    parse_wkb_linestring(io, parse_coords, crs_type; is_ring=true)
+  end
+  
+  # Always use the second branch approach as suggested in review
+  return PolyArea(exterior_ring, hole_rings...)
+end
+
+
+function parse_wkb_multipoint(io::IOBuffer, read_value::Function, crs_type)
```

Comment:
Can you please explain why the `parse_wkb_multi*` functions are still using `read_value` instead of `parse_coords` like the non-multi variants? Can you please try to be more consistent in the design?